driver-core/src/test/functional/com/mongodb/client/model/expressions/TypeExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,247 @@
+    }
+
+    // conversions
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/convert/ (38 |40)
+    // Convert is not implemented: too dynamic, conversions should be explicit.
+
+    /*
+    Useful conversions:
+    - anything-string: toString to a parsable type
+    - string-parse-anything: every type should allow parsing to a string
+      - presently excludes objects (docs) and arrays
+      - includes formatted date strings
+    - milliseconds since epoch to date
+
+    Convert also defines many conversions that do not seem useful:
+    - boolean-number conversions: t/f to-from 1/0
+    - boolean-other (oid/str/date) - always true, broken for strings; pointless
+    - number-number - "underlying" json type is changed, possible exceptions?
+     */
+
+    @Test
+    public void asStringTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/toString/
+        // asString, since toString conflicts
+        assertExpression("false", of(false).asString(), "{'$toString': [false]}");
+
+        assertExpression("1", of(1).asString());
+        assertExpression("1", of(1L).asString());
+        assertExpression("1", of(1.0).asString());
+        assertExpression("1.0", of(Decimal128.parse("1.0")).asString());
+
+        assertExpression("abc", of("abc").asString());
+
+        // this is equivalent to $dateToString
+        assertExpression("1970-01-01T00:00:00.123Z", of(Instant.ofEpochMilli(123)).asString());
+
+        // TODO:
+//        assertExpression("[]", ofIntegerArray(1, 2).asString());
+//        assertExpression("[1, 2]", ofIntegerArray(1, 2).asString());
+//        assertExpression("{a: 1}", of(Document.parse("{a: 1}")).asString());
+    }
+
+    @Test
+    public void dateAsStringTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/dateToString/
+        final Instant instant = Instant.parse("2007-12-03T10:15:30.005Z");
+        assertExpression(
+                "2007-12-03T10:15:30.005Z",
+                of(instant).asString(),
+                "{'$toString': [{'$date': '2007-12-03T10:15:30.005Z'}]}");
+
+        // with parameters
+        assertExpression(
+                "2007-12-03T05:15:30.005Z",
+                of(instant).asString(of("America/New_York"), of("%Y-%m-%dT%H:%M:%S.%LZ")),
+                "{'$dateToString': {'date': {'$date': '2007-12-03T10:15:30.005Z'}, "
+                        + "'format': '%Y-%m-%dT%H:%M:%S.%LZ', "
+                        + "'timezone': 'America/New_York'}}");
+        assertExpression(
+                "2007-12-03T14:45:30.005Z",
+                of(instant).asString(of("+04:30"), of("%Y-%m-%dT%H:%M:%S.%LZ")),
+                "{'$dateToString': {'date': {'$date': '2007-12-03T10:15:30.005Z'}, "
+                        + "'format': '%Y-%m-%dT%H:%M:%S.%LZ', "
+                        + "'timezone': '+04:30'}}");
+    }
+
+
+    // parse string
+
+    @Test
+    public void dateToStringTest() {
```

Comment:
I think that this is a matter of preference - grouping by the type of operation, versus grouping strictly by class under test.

1. This would imply that all methods on Expression should be in a single test class, but I find this unhelpful. It seems proper to test the comparison expressions together, and the type-focused expressions separately.
2. It is clearer for me to see that x-to-y and y-to-x tests directly beside each other. So they are not spread, but intentionally grouped.
3. I don't think we should move all the literal tests to `ExpressionsTest`, for example.